### PR TITLE
*: add test-smoke to the pipeline and update test-smoke

### PR DESCRIPTION
**Pull request by vivekpatani**
- add test-smoke to the tests.yaml pipeline
- add bom and dep to the test-smoke definition
- https://github.com/etcd-io/etcd/issues/17400
- https://github.com/etcd-io/etcd/issues/17401
- wait for https://github.com/etcd-io/etcd/pull/17433 to merge
- wait for https://github.com/etcd-io/etcd/pull/17434 to merge

### CODE REVIEW

**Comment by vivekpatani** [comment]
@jmhbnz @ivanvc @ahrtr PTAL

Please keep in mind, this PR should not be merged before
- https://github.com/etcd-io/etcd/pull/17433
- https://github.com/etcd-io/etcd/pull/17434

Thanks.

**Comment by ivanvc** [comment]
This looks good, as suggested by James in the triage meeting. It has a slightly different approach than how release-3.4 does it: https://github.com/etcd-io/etcd/blob/release-3.4/.github/workflows/tests.yaml#L36-L39.

Ultimately, I defer to @jmhbnz and maintainers :)

**Review by jmhbnz — APPROVED** [review]
I'm comfortable with this approach, thanks @vivekpatani.

Over time I would like to iterate on this to backport our `make verify` approach we have become accustomed to in `main` back to `release-3.5`.

**Review by jmhbnz — COMMENTED** [review]
We do still need to further investigate why the smoke job is passing despite shellcheck failing, refer: https://github.com/etcd-io/etcd/actions/runs/7925943528/job/21639943181

```bash
'shellcheck' started at Fri Feb 16 04:15:28 UTC 2024
% 'shellcheck' '-fgcc' 'build' 'test' 'scripts/build-release.sh' 'scripts/codecov_upload.sh' 'scripts/fix.sh' 'scripts/genproto.sh' 'scripts/install-marker.sh' 'scripts/release_mod.sh' 'scripts/test_images.sh' 'scripts/test_lib.sh' 'scripts/update_dep.sh' 'scripts/updatebom.sh' './build.sh' './test.sh'
FAIL: (code:1):
  % 'shellcheck' '-fgcc' 'build' 'test' 'scripts/build-release.sh' 'scripts/codecov_upload.sh' 'scripts/fix.sh' 'scripts/genproto.sh' 'scripts/install-marker.sh' 'scripts/release_mod.sh' 'scripts/test_images.sh' 'scripts/test_lib.sh' 'scripts/update_dep.sh' 'scripts/updatebom.sh' './build.sh' './test.sh'
./build.sh:25:48: warning: This { is literal. Check expression (missing ;/\n?) or quote it. [SC1083]
./build.sh:25:49: warning: This { is literal. Check expression (missing ;/\n?) or quote it. [SC1083]
./build.sh:25:58: warning: This } is literal. Check expression (missing ;/\n?) or quote it. [SC1083]
./build.sh:25:59: warning: This } is literal. Check expression (missing ;/\n?) or quote it. [SC1083]
./build.sh:32:32: note: Double quote to prevent globbing and word splitting. [SC2086]
./build.sh:33:47: note: Double quote to prevent globbing and word splitting. [SC2086]
./build.sh:34:49: note: Double quote to prevent globbing and word splitting. [SC2086]
./build.sh:35:49: note: Double quote to prevent globbing and word splitting. [SC2086]
./build.sh:36:47: note: Double quote to prevent globbing and word splitting. [SC2086]
FAIL: 'run shellcheck -fgcc build test scripts/build-release.sh scripts/codecov_upload.sh scripts/fix.sh scripts/genproto.sh scripts/install-marker.sh scripts/release_mod.sh scripts/test_images.sh scripts/test_lib.sh scripts/update_dep.sh scripts/updatebom.sh ./build.sh ./test.sh' checking failed (!=0 return code)
FAIL: 'shellcheck' failed at Fri Feb 16 04:15:32 UTC 2024
```

File: Makefile
Comment (ivanvc) [inline Makefile:158]: This should force `make` to exit with a failure if one of the passes fails, as spotted by @jmhbnz.

```suggestion
	PASSES="fmt bom dep build unit" ./test.sh 2<&1 | tee test-$(TEST_SUFFIX).log
	! grep "FAIL:" test-$(TEST_SUFFIX).log
```

Comment (serathius) [inline Makefile:158]: We no longer use `test-smoke` on main branch.
Please adopt naming and conventions from main branch.

Comment (vivekpatani) [reply Makefile:158]: @serathius the goal of this PR is to have the missing `fmt` job added back to the pipeline, there should be another over arching issue trying to closely mimic the `main` pipeline as discussed in Feb 15th 2024 Bug Triage meeting.

Please correct me if I'm wrong. @jmhbnz @ahrtr @ivanvc

Comment (vivekpatani) [reply Makefile:158]: Ack, implemented the suggested fix, thanks. @ivanvc @jmhbnz

Comment (jmhbnz) [reply Makefile:158]: Link to discussion: https://youtu.be/p7K8EH3jayo?t=1303.  This pr is the minimal pr to get basic lint checks running in `release-3.5`.  Follow-up pull requests will backport our structure of `make verify` to `release-3.5` to create parity between the two branches.

**Comment by ahrtr** [comment]
Please take a look at the workflow failures. thx

**Comment by vivekpatani** [comment]
@ahrtr rebased, the error seems to have been fixed. Thanks.

**Review by ahrtr — APPROVED** [review]

File: Makefile
Comment (ahrtr) [reply Makefile:158]: The only minor concern is the naming `test-smoke `is a little confusing. But it isn't introduced in this PR, so it's OK to keep it as it's or resolve it in a separate PR.

**Review by serathius — APPROVED** [review]

**Pull request merged**
